samples/connectivity/callnotification/src/main/java/com/example/platform/connectivity/callnotification/NotificationSource.kt

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.platform.connectivity.callnotification
+
+import android.R
+import android.annotation.SuppressLint
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.media.AudioAttributes
+import android.media.AudioManager
+import android.media.RingtoneManager
+import android.os.Build
+import androidx.annotation.RequiresApi
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationCompat.PRIORITY_MAX
+import androidx.core.app.NotificationCompat.VISIBILITY_PUBLIC
+import androidx.core.app.Person
+import androidx.core.content.getSystemService
+
+/**
+ * Source for creating notifications and notification channels
+ */
+class NotificationSource<T>(
+    private val context: Context,
+    private val notificationBroadcastClass: Class<T>,
+) {
+
+    companion object {
+
+        const val ChannelId = "1234"
+        const val ChannelIntID = 1234
+        /*
+        Notification state sent via in intent to inform receiving broadcast what action the user wants to take
+         */
+        enum class NotificationState(val value: Int) {
+            ANSWER(0),
+            REJECT(1),
+            CANCEL(2),
+            CLEARED(3);
+
+            companion object {
+                fun valueOf(value: Int) = NotificationState.values().find { it.value == value }
+            }
+        }
+
+        //Get ringtone location
+        private val ringToneUri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_RINGTONE)
+        const val NOTIFICATION_ACTION = "NOTIFICATION_ACTION"
+
+        //Channel for on going call, this has low importance so that the notification is not always shown
+        @RequiresApi(Build.VERSION_CODES.O)
+        val notificationChannelOngoing = NotificationChannel(
```

Comment:
since it's not used outside of this class make this and the other channel `private`